lib/vagrant/util/platform.rb

```diff
@@ -291,18 +293,105 @@ def wsl_path?(path)
           wsl? && !path.to_s.downcase.start_with?("/mnt/")
         end
 
+        # Compute the path to rootfs of currently active WSL.
+        #
+        # @return [String] A path to rootfs of a current WSL instance.
+        def wsl_rootfs
+          return @_wsl_rootfs if defined?(@_wsl_rootfs)
+
+          if wsl?
+            # Mark our filesystem with a temporary file having an unique name.
+            marker = Tempfile.new(Time.now.to_i.to_s)
+            logger = Log4r::Logger.new("vagrant::util::platform::wsl")
+
+            # Check for lxrun installation first
+            lxrun_path = [wsl_windows_appdata_local, "lxss"].join("\\")
+            paths = [lxrun_path]
+
+            logger.debug("checking registry for WSL installation path")
+            paths += PowerShell.execute_cmd(
+              '(Get-ChildItem HKCU:\Software\Microsoft\Windows\CurrentVersion\Lxss ' \
+                '| ForEach-Object {Get-ItemProperty $_.PSPath}).BasePath').to_s.split("\r\n").map(&:strip)
```

Comment:
Should we put a rescue in place here, in case the `execute_cmd` fails and returns nil? Otherwise we'll get an exception that `nil.map` is undefined. We could also run the command first before running map on it and adding it to paths and check the result of the Powershell command.

Reply:
The `#to_s` call will force the type so there's no need to rescue here.